api/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -486,6 +488,14 @@ type USMFeatureConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// ServiceDiscoveryFeatureConfig configures the service discovery check feature.
+type ServiceDiscoveryFeatureConfig struct {
```

Comment:
Why is this a top level feature instead of being part of USM feature?

Reply:
Service discovery is not a USM feature, it is a new product